cpp/arduino/avr/interrupt.h

```diff
@@ -0,0 +1,10 @@
+#ifndef _AVR_INTERRUPT_H_
+#define _AVR_INTERRUPT_H_
+
+// allows the production code to define an ISR method
+#define _VECTOR(N) __vector_ ## N
+#define ISR(vector, ...)            \
```

Comment:
Please add a link to the `ISR()` docs.  Also, what is the `extern` doing here?  What does this macro perform?